web/models/summary_helper.ex

```diff
@@ -0,0 +1,80 @@
+defmodule CoursePlanner.SummaryHelper do
+  @moduledoc """
+    Provides helper functions for the summary page
+  """
+  use CoursePlanner.Web, :model
+
+  alias CoursePlanner.{Repo, Terms.Term, OfferedCourse}
+
+  def get_term_offered_course_for_user(%{id: user_id, role: role}) do
```

Comment:
what do you think of adding a default parameter with the current time and then passing it down to the queries, like
```elixir
def get_term_offered_course_for_user(%{id: user_id, role: role}, current_time \\ Timex.now()) do
  case role do
    "Student" -> get_student_registered_data(user_id, current_time)
    "Teacher" -> get_teacher_registered_data(user_id, current_time)
    "Coordinator" -> get_all_terms_and_offered_courses(current_time)
    "Volunteer" -> get_all_terms_and_offered_courses(current_time)
    _           -> extract_data_from_offered_courses([])
  end
end
```
this makes the code easier to test because it doesn't depend on the real current time, and this also enable us to easily query the summary for yesterday or tomorrow if we want